File: app/src/main/java/com/projeto/testedevandroidsantander/ui/mainScreen/MainRouter.java
package com.projeto.testedevandroidsantander.ui.mainScreen;

import android.view.View;
import android.widget.AdapterView;

import java.lang.ref.WeakReference;
import java.util.Calendar;

interface MainRouterInput {
}

public class MainRouter implements MainRouterInput, AdapterView.OnItemClickListener {

    public static String TAG = MainRouter.class.getSimpleName();
    public WeakReference<MainActivity> activity;
    private Calendar currentTime;


    public Calendar getCurrentTime() {
        if(currentTime == null) return Calendar.getInstance();
        return currentTime;
    }
    public void setCurrentTime(Calendar currentTime) {
        this.currentTime = currentTime;
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

    }
}

File: Bank/app/src/main/java/com/tata/bank/main/MainRouter.kt
package com.tata.bank.main

import android.content.Intent
import com.tata.bank.login.LoginActivity
import java.lang.ref.WeakReference

interface MainRouterInput {
    fun goToLogin()

}

class MainRouter: MainRouterInput {
    lateinit var activity: WeakReference<MainActivity>

    override fun goToLogin() {
        val intent = Intent(activity.get()?.baseContext, LoginActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        activity.get()?.startActivity(intent)
    }
}